pkg/k8s/watchers/pod.go

```diff
@@ -229,6 +229,8 @@ func (k *K8sWatcher) addK8sPodV1(pod *slim_corev1.Pod) error {
 		}
 	}
 
+	k.cgroupManager.OnAddPod(pod)
```

Comment:
Should these calls be protected by a check for `option.Config.EnableSocketLBTracing`?

At least today only the socket LB tracing will use the cgroup manager. This may of course change in future, but I think it will help if this code is conditional on the feature flags that uses it.

(Same question for the initialization in the daemon, but maybe it's easier to just unconditionally initialize the object in the daemon for now and then maybe we can factor this out as a `pkg/hive:Cell` module as that framework matures)

Reply:
Given that we are expecting to extend this to be used by other features soon, it's probably not worth adding conditional checks here. We can handle this in the cgroup manager internals.